src/librustc_privacy/lib.rs

```diff
@@ -90,17 +103,24 @@ struct ReachEverythingInTheInterfaceVisitor<'b, 'a: 'b, 'tcx: 'a> {
 impl<'a, 'tcx> EmbargoVisitor<'a, 'tcx> {
     fn item_ty_level(&self, item_def_id: DefId) -> Option<AccessLevel> {
         let ty_def_id = match self.tcx.type_of(item_def_id).sty {
-            ty::Adt(adt, _) => adt.did,
-            ty::Foreign(did) => did,
-            ty::Dynamic(ref obj, ..) => obj.principal().def_id(),
-            ty::Projection(ref proj) => proj.trait_ref(self.tcx).def_id,
-            _ => return Some(AccessLevel::Public)
+            ty::Adt(adt, _) => Some(adt.did),
+            ty::Foreign(did) => Some(did),
+            ty::Projection(ref proj) => Some(proj.trait_ref(self.tcx).def_id),
+            ty::Dynamic(predicates, ..) => {
+                dyn_def_ids(self.tcx, predicates)
+                    .filter_map(|def_id| self.tcx.hir().as_local_node_id(def_id))
+                    .fold(Some(AccessLevel::Public),
```

Comment:
is this used anywhere?

Reply:
Can you use `Iterator::min`?

Reply:
Heh, it was supposed to be returned.
I'll add more tests, for all uses of `dyn_def_ids`.